euphoria-flink/src/main/java/cz/seznam/euphoria/flink/streaming/ReduceByKeyTranslator.java

```diff
@@ -1,270 +0,0 @@
-/**
- * Copyright 2016 Seznam.cz, a.s.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package cz.seznam.euphoria.flink.streaming;
-
-import com.google.common.collect.Iterables;
-import com.google.common.collect.Iterators;
-import cz.seznam.euphoria.core.client.dataset.windowing.Windowing;
-import cz.seznam.euphoria.core.client.functional.UnaryFunction;
-import cz.seznam.euphoria.core.client.operator.ReduceByKey;
-import cz.seznam.euphoria.core.client.util.Pair;
-import cz.seznam.euphoria.flink.FlinkOperator;
-import cz.seznam.euphoria.flink.functions.IteratorIterable;
-import cz.seznam.euphoria.flink.functions.PartitionerWrapper;
-import cz.seznam.euphoria.flink.streaming.windowing.FlinkWindow;
-import cz.seznam.euphoria.flink.streaming.windowing.MultiWindowedElement;
-import cz.seznam.euphoria.flink.streaming.windowing.MultiWindowedElementWindowFunction;
-import org.apache.flink.api.common.functions.ReduceFunction;
-import org.apache.flink.api.common.typeinfo.TypeInformation;
-import org.apache.flink.api.java.typeutils.ResultTypeQueryable;
-import org.apache.flink.streaming.api.datastream.DataStream;
-import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
-import org.apache.flink.streaming.api.datastream.WindowedStream;
-import org.apache.flink.streaming.api.functions.windowing.PassThroughWindowFunction;
-import org.apache.flink.streaming.api.functions.windowing.WindowFunction;
-import org.apache.flink.streaming.api.windowing.windows.Window;
-import org.apache.flink.util.Collector;
-
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.Set;
-
-class ReduceByKeyTranslator implements StreamingOperatorTranslator<ReduceByKey> {
```

Comment:
if necessary, i think we can even optimize this later directly in the api layer in `#getBasicOps` of the `RBK` operator. this would allow us to specialize the case without having to maintain separate impls in the executors.